presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -387,6 +403,9 @@ private Page extractOutput()
     {
         // INVARIANT: pagesIndex contains the full grouped & sorted data for one or more partitions
 
+        // Index is already prepared
+        windowInfo.addIndex(pagesIndex);
```

Comment:
Up to the number of ~(pages on output + number of partitions).
Each `IndexInfo` contains 2 long integers + list of integers (of size equal to number of partitions).
So, in the worst case scenario (assuming each index handles only one partition), the memory usage is equal to (2 * 8B + 4B) * number of partitions (2 longs for description of index + 1 integer for the partition size).